bots/heroes/midas_main.lua

```diff
@@ -0,0 +1,450 @@
+object.nWarpUp = 5
+object.nTransmuteUp = 10
+
+object.nGoldenSalvoUse = 15
+object.nLionsPrideUse = 5
+object.nWarpUse = 10
+
+
+
+--Midas abilities use gives bonus to harass util for a while
+function object:oncombateventOverride(EventData)
+	self:oncombateventOld(EventData)
+	
+	local addBonus = 0
+	if EventData.Type == "Ability" then
+		
+		if EventData.InflictorName == "Ability_Midas1" then  -- Using Golden Salvo will add bonus points to harass util
+			addBonus = addBonus + object.nGoldenSalvoUse
+		end
+		if EventData.InflictorName == "Ability_Midas2" then  -- Using Lion's Pride will add bonus points to harass util
+			addBonus = addBonus + object.nLionsPrideUse
+		end
+		if EventData.InflictorName == "Ability_Midas3" then  -- Using Elemental Warp will add bonus points to harass util
+			addBonus = addBonus + object.nWarpUse
+		end
+		
+	end
+	
+	if addBonus > 0 then
+		core.DecayBonus(self)
+		core.nHarassBonus = core.nHarassBonus + addBonus
+	end
+end
+object.oncombateventOld = object.oncombatevent
+object.oncombatevent 	= object.oncombateventOverride
+
+--Util calc override
+local function CustomHarassUtilityOverride(hero)
+	local nUtility = 0
+	
+	if skills.abilGoldenSalvo:CanActivate() then
+		nUtility = nUtility + object.nGoldenSalvoUp  -- Golden Salvo off CD will add bonus harass util
+	end
+	if skills.abilLionsPride:CanActivate() then
+		nUtility = nUtility + object.nLionsPrideUp   -- Lion's Pride off CD will add bonus harass util
+	end
+	if skills.abilWarp:CanActivate() then
+		nUtility = nUtility + object.nWarpUp         -- Elemental Warp off CD will add bonus harass util
+	end
+	if skills.abilTransmute:CanActivate() then
+		nUtility = nUtility + object.nTransmuteUp    -- Transmute never goes on CD -> after leveling Ultimate Midas will become more aggresive because of the stun
+	end
+	return nUtility
+end
+behaviorLib.CustomHarassUtility = CustomHarassUtilityOverride   
+
+
+----------------------------------
+--	Midas harass actions
+----------------------------------
+local function HarassHeroExecuteOverride(botBrain)
+	local bDebugEchos = false
+		
+	local unitSelf = core.unitSelf
+	local unitTarget = behaviorLib.heroTarget 
+	
+	local bActionTaken = false
+	
+	local nDistSq = Vector3.Distance2DSq(unitSelf:GetPosition(), unitTarget:GetPosition())
+	local bTargetMagicImmune = unitTarget:IsMagicImmune()
+	local vecTargetPosition = unitTarget:GetPosition()
+	local nLastHarassUtility = behaviorLib.lastHarassUtil
+	local bCanSee = core.CanSeeUnit(botBrain, unitTarget)
+	local itemPortalKey = core.GetItem("Item_PortalKey")
+	local nMyExtraRange = core.GetExtraRange(unitSelf)
+	
+	
+	--since we are using an old pointer, ensure we can still see the target for entity targeting
+	if unitTarget ~= nil then 
+		
+		-- Portal Key -- Uses PK to close in ONLY if target is NOT MAGIC IMMUNE
+		if bCanSee and not bTargetMagicImmune and itemPortalKey and itemPortalKey:CanActivate() then   
+		if nDistSq > 800 * 800 then
```

Comment:
Increase indentation after if.
This line and lines after it should be indedented more .